bsmetadata/preprocessing_utils.py

```diff
@@ -365,14 +427,42 @@ def _extract_datasource_from_url(self, url: str) -> Optional[str]:
         return parts.netloc + " > " + " > ".join(directories_parts)
 
     def preprocess(self, examples: Dict[str, List]) -> Dict[str, List]:
+        example_metadata_list = (
+            examples[self.col_to_store_metadata]
+            if self.col_to_store_metadata in examples
+            else [[] for _ in range(len(examples[self.col_url]))]
+        )
 
-        for example_url, example_meta in zip(examples["url"], examples["metadata"]):
+        # Iterate through the metadata associated with all examples in this batch.
+        for example_url, example_metadata in zip(examples[self.col_url], example_metadata_list):
             example_datasource = self._extract_datasource_from_url(example_url)
-            print(example_datasource)
```

Comment:
@chkla , nit, I've removed it (cf [this comment](https://github.com/bigscience-workshop/metadata/pull/81#discussion_r773287827))

Reply:
👍